Automattic/wp-calypso
# Multi-domain: Add error message if domain removal or add fails

Related to https://github.com/Automattic/dotcom-forge/issues/4704

## Proposed Changes

* Adds a dismissable error message to the top of the domain selection if the domain fails to be removed or added to the mini-cart due to an otherwise uncaught error.

<img width="1718" alt="Screenshot 2023-12-06 at 11 01 58 AM" src="https://github.com/Automattic/wp-calypso/assets/140841/7f5d2ebe-fcbd-4d22-b138-26799d165e31">

## Testing Instructions



* Sandbox the API and edit the API to return an error on adding or removing a domain fbhepr%2Skers%2Sjcpbz%2Schoyvp.ncv%2Serfg%2Sjcpbz%2Qwfba%2Qraqcbvagf%2Spynff.jcpbz%2Qwfba%2Qncv%2Qzr%2Qfubccvat%2Qpneg%2Qraqcbvagf.cuc%3Se%3Q8qsooq5q%23155-og
* Try adding or removing a domain from /start/domains
* Make sure /start/domains still works when there are no errors
* Make sure it works on /domains
* Make sure it looks ok on mobile



## Pre-merge Checklist
